### PR TITLE
refactoring add_cmacc function

### PR DESCRIPTION
**Pull request by dorinhogea**
Old code hiding a few bugs and a few more traps, mostly because the bloated functions are hard to parse (long live the fetch monster!).   I need this cleaned up before moving to fixing another legacy piece of code.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>8/515 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>delfile_block [setup failure]<br>sc_truncate<br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>prepare<br>ddl_csc2<br>ulonglong<br>truncatesc</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

File: db/tag.h [hunk 1]
```diff
@@ -402,10 +401,10 @@ void free_db_record(struct dbrecord *db);
 void delete_schema(const char *dbname);
 void rename_schema(const char *oldname, char *newname);
 
-void freeschema(struct schema *schema);
-void freeschema_internals(struct schema *schema);
+void freeschema(struct schema *schema, int free_indexes);
 
-struct schema *clone_schema_index(struct schema *from, int table_nmembers);
+struct schema *clone_schema_index(struct schema *from, const char *tag,
+                                  int nmembers);
```
Comment (chands10) [inline db/tag.h:407]: int nmembers should be table_nmembers

File: db/macc_glue.c [hunk 2]
```diff
@@ -460,90 +455,156 @@ static int create_key_schema(dbtable *db, struct schema *schema, int alt,
...
+        lst[(*pnlst)++] = s;
+
         rc = dyns_get_idx_tag(ix, altname, MAXTAGLEN, &where);
-        if (rc == 0 && (strcasecmp(schema->tag, ".ONDISK") == 0 ||
-                        strcasecmp(schema->tag, ".NEW..ONDISK") == 0)) {
+        if (!rc && is_ondisk) {
+            struct schema *s_alias;
             if (alt == 0) {
-                add_tag_alias(dbname, s, altname, schema->nmembers);
+                s_alias = clone_schema_index(s, altname, s->nmembers);
```
Comment (chands10) [inline db/macc_glue.c:577]: third argument should be sch->nmembers I think. Assuming sch is the table schema and s is an index schema

File: db/macc_glue.c [hunk 3]
```diff
@@ -460,90 +455,156 @@ static int create_key_schema(dbtable *db, struct schema *schema, int alt,
...
+                if (!s_alias)  {
+                    errstat_set_rcstrf(err, rc = -1, "Failed to clone tag %s",
+                                       s_alias->tag);
+                    return -1;
+                }
             } else {
+                char tmptagname[MAXTAGLEN + sizeof(".NEW.")];
                 snprintf(tmptagname, sizeof(tmptagname), ".NEW.%s", altname);
-                add_tag_alias(dbname, s, tmptagname, schema->nmembers);
+                s_alias = clone_schema_index(s, tmptagname, s->nmembers);
```
Comment (chands10) [inline db/macc_glue.c:586]: Also should be sch->nmembers

File: db/macc_glue.c (hunk 2 above)
Comment (dorinhogea) [reply db/macc_glue.c:577]: oh yes, this is the datacopy length for the index, and here I think we create a new tagname for it..

File: db/tag.h (hunk 1 above)
Comment (dorinhogea) [reply db/tag.h]: renamed and fixed

File: db/macc_glue.c (hunk 2 above)
Comment (dorinhogea) [reply db/macc_glue.c]: renamed and fixed

File: db/macc_glue.c (hunk 3 above)
Comment (dorinhogea) [reply db/macc_glue.c]: renamed and fixed

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>6/516 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>electbug2 [setup failure]<br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>prepare<br>ddl_csc2<br>ulonglong</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>9/517 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover_noroll1_generated<br>yast_stat4scan_generated<br>yast<br>analyze_partial_index_off_generated<br>analyze<br>read_committed<br>ddl_analyze<br>tunables<br>truncatesc_ufid_on_generated</p>

**Review by mponomar — APPROVED** [review]

**Comment by dorinhogea** [comment]
@mponomar thank you!
